astropy/io/fits/hdu/image.py

```diff
@@ -928,8 +928,8 @@ def __init__(self, data=None, header=None, do_not_scale_image_data=False,
     def match_header(cls, header):
         card = header.cards[0]
         return (card.keyword == 'SIMPLE' and
-                ('GROUPS' not in header or header['GROUPS'] != True) and
-                card.value == True)
+                ('GROUPS' not in header or header['GROUPS'] is not True) and
```

Comment:
Ohh, after having an actual look at the standard, this should have been OK as the value should have been boolean. 
My resolution is, never to make any stylistic change to the fits module again to avoid braking what's working fine.